app/cp_nigeria/helpers.py

```diff
@@ -186,3 +194,357 @@ def create_report_content(self, project):
+        system_params["label"] = system_params["supply_source"] + "_" + system_params["category"]
+        # TODO include excess generation (to be used by excess gen tariff - also not prio while not considering feedin)
+        return system_params
+
+    def collect_financial_params(self):
+        """
+        This method collects the financial parameters from the economic and the equity data set on the project.
+        """
+        qs_eq = EquityData.objects.filter(scenario=self.project.scenario).values(
+            "debt_start",
+            "fuel_price_increase",
+            "grant_share",
+            "debt_share",
+            "debt_interest_MG",
+            "debt_interest_SHS",
+            "equity_interest_MG",
+            "equity_interest_SHS",
+        )
+        qs_ed = EconomicData.objects.filter(project=self.project).values("discount", "tax")
+        financial_params = pd.concat([pd.DataFrame.from_records(qs_eq), pd.DataFrame.from_records(qs_ed)], axis=1)
+
+        return financial_params
+
+    @staticmethod
+    def yearly_increase(base_amount, growth_rate, year):
+        return base_amount * (1 + growth_rate) ** year
+
+    def growth_over_lifetime_table(self, df, base_col, growth_col, index_col=None):
+        """
+        This method creates a wide table over the project lifetime based on start values and growth rate of the given
+        parameters. It is used to calculate both the system growth over the project lifetime and the growth in
+        costs/revenues based on a given annual increase rate (that can be set independently for each parameter in
+        collect_system_params() and cost_assumptions.csv). This method is used in the revenue and losses_over lifetime
+        methods.
+        """
+        lifetime_df = pd.DataFrame(
+            {
+                self.project_start + year: self.yearly_increase(df[base_col], df[growth_col], year)
+                for year in range(self.project_duration)
+            }
+        )
+
+        if index_col is not None:
+            if isinstance(index_col, list):
+                lifetime_df.index = [np.array(df[col]) for col in index_col]
+            else:
+                lifetime_df.index = df[index_col]
+
+        return lifetime_df
+
+    def add_diff_rows(self, df):
+        """
+        This method adds new consumers rows over project lifetime both for mini-grid and SHS consumers. This is needed
+        to calculate the connection fees for new consumers per project year, given a yearly consumer increase. So far,
+        consumer increase is not accounted for, so all but the initial year should return 0.
+        """
+        vars = [ix for ix in df.index if "nr_consumers" in ix]
+        for var in vars:
+            diff = df.T[var].diff()
+            diff[self.project_start] = df[self.project_start].loc[var]
+            new_var = f"{var}_new"
+            df.loc[new_var] = diff
+
+        return df
+
+    def calculate_capex(self):
```

Comment:
```suggestion
@property
    def calculate_capex(self):
```

This allows to call `self.calculate_capex` instead of `self.calculate_capex()`. We could also rename the method simply `def capex(self)`

Reply:
This comment is valid for all other method without any argument